src/js/edu-benefits/definitions/educationProgram.js

```diff
@@ -2,31 +2,18 @@ import _ from 'lodash/fp';
 
 import { showSchoolAddress } from '../utils/helpers';
 import * as address from '../../common/schemaform/definitions/address';
-import educationTypeUISchema from '../definitions/educationType';
+import educationTypeUISchema from './educationType';
 
 export const uiSchema = {
   'ui:order': ['name', 'educationType', 'address'],
-  address: address.uiSchema(),
+  address: _.merge(address.uiSchema(), {
```

Comment:
Thanks for the thorough explanation.

Tragically, the elegant `hideIf()` solution makes the unit test run ever so slightly more slowly (could not get the address inputs to show up after changing type to `college` without a timeout 😢 ). I'll comment out the timeout to avoid flakiness, but we may want to discuss solutions at the next FE architecture sync.